### PR TITLE
separate tasks to power off nodes with ipmi and redfish

### LINKED ISSUE #155: installer role powers off nodes only via IPMI
The [node_prep](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/roles/node_prep/tasks/10_validation.yml#L483-L524) role generates two roles dell_hosts_redfish and hp_hosts_redfish, but to power off hosts it only uses IPMI in the [installer](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/roles/installer/tasks/59_power_off_cluster_servers.yml) role, then it is required to have it enabled. 

Ideally if there are nodes in the dell_hosts_redfish and hp_hosts_redfish they should use a Redfish call to power off servers, then no need for IPMI.

### PR DESCRIPTION
**Pull request by manurodriguez**
Fixes #155

### CODE REVIEW

**Comment by manurodriguez** [comment]
check dallas ocp-4.14-vanilla

**Comment by manurodriguez** [comment]
check dallas ocp-4.14-vanilla

File: roles/installer/tasks/59_power_off_cluster_servers.yml [hunk 1]
```diff
@@ -26,3 +29,21 @@
...
+    username: "{{ hostvars[item]['ipmi_user'] }}"
+    password: "{{ hostvars[item]['ipmi_password'] }}"
+  when:
+    - ( hostvars[item]['poweroff'] is not defined )
+      or ( hostvars[item]['poweroff']|bool | default(True) )
+    - ( groups['dell_hosts_redfish'] is defined and groups['dell_hosts_redfish'] | length > 0 )
+      or ( groups['hp_hosts_redfish'] is defined and groups['hp_hosts_redfish'] | length > 0 )
+  with_items:
+    - "{{ groups['dell_hosts_redfish'] | default([]) }}"
+    - "{{ groups['hp_hosts_redfish'] | default([]) }}"
```
Comment (tonyskapunk) [inline roles/installer/tasks/59_power_off_cluster_servers.yml:47]: nit:
```suggestion
  loop: "{{ groups['dell_hosts_redfish'] | default([]) + groups['hp_hosts_redfish'] | default([]) }}"
```

**Review by ramperher — COMMENTED** [review]
Looks like libvirt job got stuck before reaching the task under test, can we repeat that test to confirm it's working?

**Comment by manurodriguez** [comment]
> Looks like libvirt job got stuck before reaching the task under test, can we repeat that test to confirm it's working?

Hi Ramon, good observation, I checked the libvirt job and actually what it got stuck was the output in the DCI console, but if you look at the [ansible.log](https://www.distributed-ci.io/files/59138340-ceb2-4290-a2ed-b2b06e86c268) file all it's in there[1]. So I think it's more like a DCI issue, because it's not showing anything after the task `Check if override path is defined for customize_filesystem` and that's like 20 tasks before the task that powers off the nodes. But again the full output it's on the ansible.log

[1]
```
2024-02-01 15:26:11,664 p=1161722 u=dci-openshift-agent n=ansible | TASK [redhatci.ocp.installer : Power off hosts via IPMI] ***********************
2024-02-01 15:26:12,216 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-master-0) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-master-0", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:12,591 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-master-1) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-master-1", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:13,052 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-master-2) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-master-2", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:13,446 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-worker-0) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-worker-0", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:13,829 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-worker-1) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-worker-1", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:14,210 p=1161722 u=dci-openshift-agent n=ansible | ok: [provisionhost.server09.partnerci.bos2.lab] => (item=dciokd-worker-2) => {"ansible_loop_var": "item", "attempts": 1, "changed": false, "item": "dciokd-worker-2", "powerstate": "off", "warnings": ["The value False (type bool) in a string field was converted to 'False' (type string). If this does not look like what you expect, quote the entire value to ensure it does not change."]}
2024-02-01 15:26:14,210 p=1161722 u=dci-openshift-agent n=ansible 

```

**Review by ramperher — APPROVED** [review]
Clear for me, thanks!

File: roles/installer/tasks/59_power_off_cluster_servers.yml [hunk 2]
```diff
@@ -26,3 +29,19 @@
...
+    category: Systems
+    command: PowerGracefulShutdown
+    baseuri: "{{ hostvars[item]['ipmi_address'] }}"
+    username: "{{ hostvars[item]['ipmi_user'] }}"
+    password: "{{ hostvars[item]['ipmi_password'] }}"
+  when:
+    - ( hostvars[item]['poweroff'] is not defined )
+      or ( hostvars[item]['poweroff']|bool | default(True) )
+    - ( groups['dell_hosts_redfish'] is defined and groups['dell_hosts_redfish'] | length > 0 )
+      or ( groups['hp_hosts_redfish'] is defined and groups['hp_hosts_redfish'] | length > 0 )
```
Comment (tkrishtop) [inline roles/installer/tasks/59_power_off_cluster_servers.yml:44]: We could maybe rewrite it a bit nicer 
```
when: |
  (
    (hostvars[item]['poweroff'] is not defined) or
    (hostvars[item]['poweroff'] | bool | default(True))
  ) and (
    (
      groups['dell_hosts_redfish'] is defined and
      groups['dell_hosts_redfish'] | length > 0
    ) or (
      groups['hp_hosts_redfish'] is defined and
      groups['hp_hosts_redfish'] | length > 0
    )
  )
```

**Review by tkrishtop — APPROVED** [review]
LGTM, one minor formatting suggestion, please feel free to ignore it

**Review by tonyskapunk — APPROVED** [review]
LGTM

**Pull request merged**